packages/cli/core/src/commands/ui/deploy.spec.ts

```diff
@@ -465,6 +473,69 @@ describe('ui:deploy', () => {
         }
       );
 
+    test
+      .stdout()
+      .stderr()
+      .do(() => {
+        doMockConfirm();
+        mockedConfirm.mockResolvedValueOnce(true);
+        mockHostedPageList.mockResolvedValueOnce({
+          items: [{name: validJsonConfig.name, id: 'thisistheid'}],
+        });
+      })
+      .command(['ui:deploy'])
+      .it(
+        'should call hostedPages.list when no page id argument is passed and a page with the same already exists',
+        () => {
+          expect(mockHostedPageList).toHaveBeenCalledWith(
+            expect.objectContaining({
+              filter: validJsonConfig.name,
+              perPage: 1000,
+            })
+          );
+        }
+      );
+
+    test
+      .stdout()
+      .stderr()
+      .do(() => {
+        doMockConfirm();
+        mockedConfirm.mockResolvedValueOnce(true);
+        mockHostedPageList.mockResolvedValueOnce({
+          items: [{name: validJsonConfig.name, id: 'thisistheid'}],
+        });
+      })
+      .command(['ui:deploy'])
+      .it(
+        'should call hostedPages.update when no page id argument is passed, a page with the same already exists, and the user confirms the overwrite',
+        () => {
+          expect(mockHostedPageUpdate).toHaveBeenCalledWith({
+            ...expectedHostedPage,
+            id: 'thisistheid',
```

Comment:
extract into a `testId` const e.g. for reusability ?

Reply:
- [ ] refactor id